crates/bevy_ecs/src/system/combinator.rs

```diff
@@ -421,8 +421,30 @@ where
         &mut self,
         world: UnsafeWorldCell,
     ) -> Result<(), SystemParamValidationError> {
-        // SAFETY: Delegate to other `System` implementations.
-        unsafe { self.a.validate_param_unsafe(world) }
+        // SAFETY: Delegate to the `System` implementation for `a`.
+        let a_validation = unsafe { self.a.validate_param_unsafe(world) };
+
+        // SAFETY: Delegate to the `System` implementation for `b`.
+        let b_validation = unsafe { self.b.validate_param_unsafe(world) };
+
+        match (a_validation, b_validation) {
+            (Ok(()), Ok(())) => Ok(()),
+            (Ok(_), Err(err_b)) => Err(err_b),
+            (Err(err_a), Ok(_)) => Err(err_a),
+            (Err(err_a), Err(err_b)) => {
+                // Combine the errors from both systems, using the most severe outcome.
```

Comment:
We decided to short circuit for tuples, per https://github.com/bevyengine/bevy/pull/18504#discussion_r2011030275.  Why not short circuit here, as well?  That's a shorter implementation (using `?`), more consistent with multi-parameter systems, and doesn't allocate strings.